packages/fields/src/fields/author/author-view.tsx

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { commentAuthorAvatar as authorIcon } from '@wordpress/icons';
+import { __experimentalHStack as HStack, Icon } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import type { User } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import type { BasePost } from '../../types';
+
+function AuthorView( { item }: { item: BasePost } ) {
+	const { text, imageUrl } = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			let user: User | undefined;
+			if ( !! item.author ) {
+				user = getEntityRecord( 'root', 'user', item.author );
+			}
+			return {
+				imageUrl: user?.avatar_urls?.[ 48 ],
+				text: user?.name,
+			};
```

Comment:
I had to rewrite the nice-looking dynamic `getUser` selector into using `getEntityRecord` one. TypeScript was complaining because it wasn't able to find the `getUser` one. It was like this before:

```js
const { getUser } = select( coreStore );
			const user = getUser( item.author );
			return {
				imageUrl: user?.avatar_urls?.[ 48 ],
				text: user?.name,
			};
```

cc @youknowriad @sirreal can we use the dynamic selectors in TypeScript? I didn't find how.

Reply:
I'd love to help but honestly I'm not up to speed on the data packages and TypeScript. I believe @adamziel was working on TypeScript support in the past but I'm not sure what the current status is.

This PR may be good to look at: https://github.com/WordPress/gutenberg/pull/43643